shared/actions/unlock-folders.js

```diff
@@ -0,0 +1,39 @@
+/* @flow */
+
+import * as Constants from '../constants/unlock-folders'
+import {routeAppend} from '../actions/router'
+
+import type {RouteAppend} from '../constants/router'
+import type {TypedAsyncAction} from '../constants/types/flux'
+import type {LoadDevices, ToPaperKeyInput, CheckPaperKey, Finish} from '../constants/unlock-folders'
+
+// TODO this won't work in a remote component until we have a synchronous way to get the latest state in a remote component.
+
+function nextPhase (): TypedAsyncAction<RouteAppend> {
+  return (dispatch, getState) => {
+    // TODO careful here since this will not be sync on a remote component!
+    const phase: string = getState().unlockFolders.phase
```

Comment:
i find this model problematic. we're doing this in signup and i think its strange the the value in the store is disconnected from what we see. maybe i'm doing something wrong but what i see is phase = '1' and its not on screen '1' until some action decides to call this to make the ui reflect the store. am i misunderstanding this